src/vs/workbench/services/path/browser/pathService.ts

```diff
@@ -9,15 +9,26 @@ import { IPathService, AbstractPathService } from 'vs/workbench/services/path/co
 import { URI } from 'vs/base/common/uri';
 import { Schemas } from 'vs/base/common/network';
 import { IWorkbenchEnvironmentService } from 'vs/workbench/services/environment/common/environmentService';
+import { IWorkspaceContextService } from 'vs/platform/workspace/common/workspace';
 
 export class BrowserPathService extends AbstractPathService {
 
 	constructor(
 		@IRemoteAgentService remoteAgentService: IRemoteAgentService,
-		@IWorkbenchEnvironmentService environmentService: IWorkbenchEnvironmentService
+		@IWorkbenchEnvironmentService private readonly environmentService: IWorkbenchEnvironmentService,
+		@IWorkspaceContextService private readonly contextService: IWorkspaceContextService
 	) {
 		super(URI.from({ scheme: Schemas.vscodeRemote, authority: environmentService.configuration.remoteAuthority, path: '/' }), remoteAgentService);
 	}
+
+	defaultUriScheme(): string {
+		if (this.environmentService.configuration.remoteAuthority) {
+			return Schemas.vscodeRemote;
+		} else {
+			// Empty workspace is not supported, so we will always have at least one folder.
```

Comment:
Should we add an assert here that throws if the workspace is empty so that we at least get an error with a meaningful message?

Reply:
Added!